### DISCUSSION TITLE
API troubles list issues

### DISCUSSION THREAD

**Discussion by joan-saum** [community/community#24893] [Apps, API and Webhooks]
<p>Hello!</p>
<p>I’m using the GitHub API in one of my project since a few months, and i’m dealing with a problem i never seen before</p>
<p>Sometimes, this request : “<a href="https://api.github.com/repos/Homebrew/homebrew-cask/issues?per_page=1&amp;state=all" rel="nofollow noopener">https://api.github.com/repos/Homebrew/homebrew-cask/issues?per_page=1&amp;state=all</a>” don’t return me anything. I have no answer at all</p>
<p>It seems to happen only on the repository Homebrew/homebrew-cask, i never get it with another repository</p>
<p>We can reproduce it by using this commands in the terminal :</p>
<pre><code>curl -X GET "https://api.github.com/repos/Homebrew/homebrew-cask/issues?per_page=1&amp;state=all"
</code></pre>
<p>And run it many times until get a empty answer</p>
<p>Hope someone can find where are the troubles :slight_smile:</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/joan-saum">@joan-saum</a>,</p>
<p>When I try with different state parameters, I get a result with “open” but not with “closed”. It is possible that the API is having an issue with the number of closed issues in that repository or maybe there’s an issue with a particular issue or pull request. I’ll let our product team know and see if they have any further insight.</p>
<p>Thanks!</p>

**Comment by joan-saum** [comment]
<p>Thanks for you reply <a class="mention" href="/that-pat">@that-pat</a>, let me know if you get some news !</p>

**Comment by lee-dohm** [comment]
<p>Are you certain that you aren’t running into <a href="https://developer.github.com/v3/#rate-limiting" rel="nofollow noopener">API rate limits</a>? The reason why I ask is because asking for only one issue per page means that you’re going to run into your rate limit 30 times faster than not including the <code>per_page</code> qualifier.</p>

**Comment by joan-saum** [comment]
<p>Hello <a class="mention" href="/lee-dohm">@lee-dohm</a></p>
<p>My soft used the <code>per_page</code> qualifier because i’m trying to know the number of issues.</p>
<p>So my logic is : i call only one time the road /issues?per_page=1 and i look at the variable LastPage the number. Because there is only 1 issue per page, the LastPage indicate the number of issues. I don’t iterate on all issues</p>
<p>Usually, if i run into API rate limits, i got an explicit error message.</p>
<p>I just tried 20 min ago to run many times the command:</p>
<pre><code>curl -X GET "https://api.github.com/repos/Homebrew/homebrew-cask/issues?per_page=1&amp;state=all
</code></pre>
<p>I run this command maybe around 40 times (manually, without a script).</p>
<p>I got one time this answer :</p>
<pre><code>{
  "message": "Server Error"
}
</code></pre>
<p>And one other time, i got no answer at all, like what i describe in my first post.</p>
<p>The others answers was good. So, very rarely there is an error</p>

**Comment by lee-dohm** [comment]
<p>Do you get the same error when you don’t include the <code>per_page</code> qualifier? Because it could be that the query you’re issuing is causing unexpected stress on a large repo. If you use the API more the way it is intended, it may work more reliably.</p>
<p>You would only have to issue two queries to get the same information. One to get the first page, which gives you the last page URL, then another to get the last page of issues. Count the number of pages, multiply by the number of items per page (defaults to 30), plus the number of items on the last page. That gives you the count of issues in two queries.</p>

**Comment by joan-saum** [comment]
<p>Yes</p>
<p>Actually, i tried many times this request (list all issues) :</p>
<pre><code>curl -X GET "https://api.github.com/repos/Homebrew/homebrew-cask/issues?per_page=1&amp;state=all"
</code></pre>
<p>With the<code>per_page</code> qualifier and without also. No problems</p>
<p>And then i tried this request (list only closed issues):</p>
<pre><code>curl -X GET "https://api.github.com/repos/Homebrew/homebrew-cask/issues?state=closed"
</code></pre>
<p>Again, with the<code>per_page</code> qualifier and without. And i got the error i told before (One error is Server error, the other is just no answer at all)</p>
<p>I made a screenshot of my terminals, so you can also see the exact time of the request and your technical team can see the logs at this time</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9a4c00daa6b475ae824f2c302e4cee2446aa1d5c.png" data-download-href="/uploads/short-url/m0YmK9K7QioaL8rvwk4vcddNKqM.png?dl=1" title="Capture du 2018-10-29 09-44-29.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16139407/181087402-12fc0b76-f530-4723-832f-d367a5a918bd.png" alt="Capture du 2018-10-29 09-44-29.png" data-base62-sha1="m0YmK9K7QioaL8rvwk4vcddNKqM" width="690" height="200" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/9a4c00daa6b475ae824f2c302e4cee2446aa1d5c_2_690x200.png, https://aws1.discourse-cdn.com/github/optimized/2X/9/9a4c00daa6b475ae824f2c302e4cee2446aa1d5c_2_1035x300.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/9/9a4c00daa6b475ae824f2c302e4cee2446aa1d5c_2_1380x400.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9a4c00daa6b475ae824f2c302e4cee2446aa1d5c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture du 2018-10-29 09-44-29.png</span><span class="informations">1473×428 75.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>PS : The text before the request is the answer of github API with the same request. It show that sometimes it work.</p>

**Comment by joan-saum** [comment]
<p>Hi <a class="mention" href="/that-pat">@that-pat</a></p>
<p>Any news from your product team about this error ?</p>

**Comment by ThirdEyeRose** [comment]
<p><a class="mention" href="/joan-saum">@joan-saum</a>, nothing definitive, unfortunately. They did experience an outage around the time that you asked this question which could have impacted the service in question. Have you gotten different results when trying again?</p>

**Comment by joan-saum** [comment]
<p>Yes i did</p>
<p>You can see my conversation with lee-dohm below. I tried monday again and i got some problems to list closed issue. I also post a screenshot of my terms so the tech team can see the exact time of my requests</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p><a class="mention" href="/joan-saum">@joan-saum</a>, hm. That’s good information to have. Unfortunately, I think that might be beyond my level of expertise for troubleshooting this issue in a forum setting. Have you contacted <a href="https://github.com/contact" rel="nofollow noopener">private support</a>? If you haven’t, you may want to so that we can do some advanced troubleshooting with you. You can submit a link to this forum post to prevent having to repeat yourself.</p>

**Reply by dollar159** [reply]
it's the whole android system not updating some else has put restrictions on android device I've tried to get in touch with security but who ever this is,no doubt there from Android developer this is my first to get through to someone
